lib/src/facts/aix/networking_resolver.cc

```diff
@@ -0,0 +1,239 @@
+#include <internal/facts/aix/networking_resolver.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <unordered_map>
+
+#include <inttypes.h>
+#include <net/if.h>
+#include <net/if_dl.h>
+#include <net/route.h>
+#include <netinet/in.h>
+#include <sys/ndd_var.h>
+#include <sys/kinfo.h>
+
+// This usage is recommended in several mailing lists, and is used at
+// least in Samba to query mac addresses. I saw some references to old
+// IBM publications also recommending its use, but couldn't actually
+// dig those pubs up.
+//
+// The structs it *returns* are in headers, though (specifically
+// mentioning usage of this function).
+//
+// This is all leftovers from ancient versions of the BSD networking
+// stack that every OS in the world has stolen/borrowed. It is pretty
+// gross in a lot of ways, but is what we have to work with on AIX.
+//
+// There does not appear to be a different or documented way to get
+// this information, outside of shelling out to tools which provide
+// human- rather than machine-readable output.
+extern "C" int getkerninfo(int, char*, int*, int32long64_t);
+
+template <typename T>
+static std::vector<T> getkerninfo(int query) {
+    for (;;) {
```

Comment:
We could bound it in case the size is ever increasing, but it'll error with out-of-memory anyway. The similar example I was thinking of is https://github.com/puppetlabs/facter/blob/master/lib/src/facts/windows/networking_resolver.cc#L154-L169.